### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Ostyk**
**FIXED:**
- fixed class hardcoding (all models) [DONE]
- add categories numbers from dataset [DONE]
- manage label file name (reading dataset coco.py) [DONE]
- logging metrics [DONE] (Logging 8 config metrics + 5 result based metrics to CSV in results folder)
- interpolation names [DONE]
- testing check for CUDA (change nnet.cuda to device when testing on CPU)
- handle two datasets in same folder [DONE]
- write script to test all posible variants (changing two fields in config .json - categories & interpolation_mode) [Done][Not tested]
- fix coco stats csv fields [DONE]
- fix max iter [DONE]
- fix test set reading (image no shape) [DONE]
- test and valid pkl cache fix(different datasets) [DONE]

**TO DO/future work:**

- train on one interp method, test on other?
- DLA34 has interpolation class and mode but it's not used
- TensorRT
- no_flip when testing
- Cpools on CPU (windows or linux)
- gitignore of results/models
- AR interpretation: 
- maybe resource script should be a wrapper
- think of warm up time (should not count towards FPS on edge devices) → should not be a factor
- make sure testing without CUDA works in terms of nnet.cuda-logging overfitting/ losses etc → TO DO: later
- FIX TRAINING CACHE (delete before training)
- fix none in results folder [TO DO]
-

### CODE REVIEW

**Review by Ostyk — COMMENTED** [review]
LGTM

**Comment by Ostyk** [comment]
sorry
